internal/graph/check_test.go

```diff
@@ -2998,3 +3000,329 @@ func TestIteratorReadStartingFromUser(t *testing.T) {
 		})
 	}
 }
+
+func TestBuildTupleKeyConditionFilter(t *testing.T) {
+	tests := []struct {
+		name         string
+		tupleKey     *openfgav1.TupleKey
+		model        *openfgav1.AuthorizationModel
+		context      map[string]interface{}
+		conditionMet bool
+		expectedErr  error
+	}{
+		{
+			name:     "no_condition",
+			tupleKey: tuple.NewTupleKey("document:1", "can_view", "user:maria"),
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+
+				type user
+
+				type document
+					relations
+						define can_view: [user]`),
+			context:      map[string]interface{}{},
+			conditionMet: true,
+			expectedErr:  nil,
+		},
+		{
+			name:     "condition_not_met",
+			tupleKey: tuple.NewTupleKeyWithCondition("document:1", "viewer", "user:maria", "correct_ip", nil),
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+
+				type user
+
+				type document
+					relations
+						define can_view: [user]`),
+			context:      map[string]interface{}{},
+			conditionMet: false,
+			expectedErr:  condition.NewEvaluationError("correct_ip", fmt.Errorf("condition was not found")),
+		},
+		{
+			name:     "condition_missing_parameter",
+			tupleKey: tuple.NewTupleKeyWithCondition("document:1", "can_view", "user:maria", "correct_ip", nil),
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+
+				type user
+
+				type document
+					relations
+						define can_view: [user]`),
+			context:      map[string]interface{}{},
+			conditionMet: false,
+			expectedErr:  condition.NewEvaluationError("correct_ip", fmt.Errorf("condition was not found")),
+		},
+		{
+			name:     "condition_missing_parameter",
+			tupleKey: tuple.NewTupleKeyWithCondition("document:1", "can_view", "user:maria", "x_y", nil),
+			model: parser.MustTransformDSLToProto(`
+				model
+					schema 1.1
+
+				type user
+
+				type document
+					relations
+						define can_view: [user with x_y]
+
+				condition x_y(x: int, y: int) {
+					x == 1 && y == 0
+				}
+`),
+			context:      map[string]interface{}{"x": 5},
+			conditionMet: false,
+			expectedErr: condition.NewEvaluationError("x_y",
+				fmt.Errorf("tuple 'document:1#can_view@user:maria' is missing context parameters '[y]'")),
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			ts, err := typesystem.NewAndValidate(context.Background(), tt.model)
+			require.NoError(t, err)
+
+			contextStruct, err := structpb.NewStruct(tt.context)
+			require.NoError(t, err)
+
+			iterFunc := buildTupleKeyConditionFilter(context.Background(), contextStruct, ts)
```

Comment:
Not for this PR, but: i am starting to feel like this file will become huge with tests. That is not a bad thing, but here we are testing a function that is not strictly related to Check and could be used in other APIs.

I feel like we need a `check_utils` file or something where all the auxiliary functions could live.

Reply:
Absolutely, we can move the function that are not direct local checkers into this check_utils function

Reply:
I probably want to do this after the next PR (where it is nothing but move functions out into the util file)

Reply:
perhaps a new package is what we should aim for? something like `checkhelper`?

Reply:
however a more meaningful package would be ideal

Reply:
We can have something like checkUtil package and put all the util function there.

Reply:
`buildTupleKeyConditionFilter` can be re-used in Check, ListObjects and Listusers, and we can fix this: https://github.com/openfga/openfga/issues/1567

I'd move to the `storage` folder, maybe inside `iterators`